### ISSUE TITLE
Getting "loop Promises must not throw!" error when running the example

### ISSUE THREAD

**Issue by danielmohacsi** [raisemarketplace/redux-loop#24]
Hi there,

First of all, I would like to thank you for this really awesome library!

I have started playing around with the example. The `delay` function rejects the promise with a 20% chance. If the promise gets rejected, the error gets printed to the console.

I am considering to choose this framework to manage my async flow as I'm porting my third-party JS application from Flux to Redux.
I can work this around by ensuring that my promises won't throw, but I cannot use this awesome library if it prints errors to the console. That's strictly forbidden in my third-party application.

**Comment by lukewestby** [comment]
Thanks for reporting this, and thank you for your interest @danielmohacsi!

Have you changed any of the example code or does it show this message just as it is?

**Comment by lukewestby** [comment]
@danielmohacsi I'm not seeing this happen in the example when installed without modifications from this repo. Did you remove or forget a `.catch()` call on any of your effect promises?

**Comment by danielmohacsi** [comment]
Thanks for getting back to me!

You're right, I have modified the example a little bit. I have changed the ```const Api (...)``` definition to the following one:

```JavaScript
const Api = {
  shortIncrement: (amount) => (
    delay(10)
      .then(() => Actions.shortIncrementSucceed(amount))
      .catch((err) => {
        console.log(err);  // Logs undefined - OK
        Actions.shortIncrementFail();
      })
  ),

  longIncrement: (amount) => (
    delay(3000)
      .then(() => Actions.longIncrementSucceed(amount))
      .catch(() => Actions.longIncrementFail())
  )
};
```

Sorry for not stating this in the issue description!

**Comment by danielmohacsi** [comment]
And I found it, I'm not returning any action. Sorry, I'm an idiot.

**Comment by lukewestby** [comment]
@danielmohacsi even though it turned out that there wasn't a bug in the example or library _functionality_, I think we can consider this a bug in the _error message_ in that it wasn't helpful enough to point out where the issue was. I think we can reopen this with the `bug` tag and I'll improve the error message to include this case where an action wasn't being returned as well as some possible other cases.

Thanks for your help finding this!

**Comment by danielmohacsi** [comment]
That's true, but you've written your comments to the example, I just did not read it.

Anyway, thanks for your help!

**Comment by lukewestby** [comment]
Fixed in v2.0.0!
